tests/toolkit/matl_buy_policy_tests.cc

```diff
@@ -226,25 +277,239 @@ TEST_F(MatlBuyPolicyTests, ActiveDormantMultipleCycles) {
-  EXPECT_EQ(11, third_cycle);
+  EXPECT_EQ(7, third_cycle);
+
+  delete a;
+}
+
+TEST_F(MatlBuyPolicyTests, UniformActiveDormant) {
+  using cyclus::QueryResult;
+
+  boost::shared_ptr<UniformIntDist> a_dist = boost::shared_ptr<UniformIntDist>(new UniformIntDist(2, 4));
+  boost::shared_ptr<UniformIntDist> d_dist = boost::shared_ptr<UniformIntDist>(new UniformIntDist(1, 2));
+  
+  int dur = 10;
+  double throughput = 1;
+
+  cyclus::MockSim sim(dur);
+  cyclus::Agent* a = new TestFacility(sim.context());
+  sim.context()->AddPrototype(a->prototype(), a);
+  sim.agent = sim.context()->CreateAgent<cyclus::Agent>(a->prototype());
+  sim.AddSource("commod1").Finalize();
+
+  TestFacility* fac = dynamic_cast<TestFacility*>(sim.agent);
+
+  cyclus::toolkit::ResBuf<cyclus::Material> inbuf;
+  cyclus::toolkit::MatlBuyPolicy policy;
+  policy.Init(fac, &inbuf, "inbuf", throughput, a_dist, d_dist, NULL)
+        .Set("commod1").Start();
 
-  // confirm total inbuf recieved material
-  EXPECT_DOUBLE_EQ(7, inbuf.quantity());
+  EXPECT_NO_THROW(sim.Run());
+
+  QueryResult qr = sim.db().Query("Transactions", NULL);
+  // confirm that transactions are only occurring during active periods
+  int first_cycle = qr.GetVal<int>("Time", 0);
+  EXPECT_EQ(0, first_cycle);
+  // second cycle should include time 3
+  int second_cycle = qr.GetVal<int>("Time", 2);
+  EXPECT_EQ(3, second_cycle);
+  // third cycle should include 8
+  int third_cycle = qr.GetVal<int>("Time", 5);
+  EXPECT_EQ(8, third_cycle);
 
   delete a;
 }
 
+TEST_F(MatlBuyPolicyTests, NormalActiveDormant) {
+  using cyclus::QueryResult;
+
+  boost::shared_ptr<NormalIntDist> a_dist = 
+  boost::shared_ptr<NormalIntDist>(new NormalIntDist(5, 1, 2, 10));
+  boost::shared_ptr<NormalIntDist> d_dist = boost::shared_ptr<NormalIntDist>(new NormalIntDist(3, 0.5, 1, 5));
+  
+  int dur = 25;
+  double throughput = 1;
+
+  cyclus::MockSim sim(dur);
+  cyclus::Agent* a = new TestFacility(sim.context());
+  sim.context()->AddPrototype(a->prototype(), a);
+  sim.agent = sim.context()->CreateAgent<cyclus::Agent>(a->prototype());
+  sim.AddSource("commod1").Finalize();
+
+  TestFacility* fac = dynamic_cast<TestFacility*>(sim.agent);
+
+  cyclus::toolkit::ResBuf<cyclus::Material> inbuf;
+  cyclus::toolkit::MatlBuyPolicy policy;
+  policy.Init(fac, &inbuf, "inbuf", throughput, a_dist, d_dist, NULL)
+        .Set("commod1").Start();
+
+  EXPECT_NO_THROW(sim.Run());
+
+  QueryResult qr = sim.db().Query("Transactions", NULL);
+  // confirm that transactions are only occurring during active periods
+  int first_cycle = qr.GetVal<int>("Time", 0);
+  EXPECT_EQ(0, first_cycle);
+  // second cycle should include on time 9
+  int second_cycle = qr.GetVal<int>("Time", 6);
+  EXPECT_EQ(9, second_cycle);
+  // third cycle should include time 19
+  int third_cycle = qr.GetVal<int>("Time", 13);
+  EXPECT_EQ(19, third_cycle);
+
+  delete a;
+}
+
+// TEST_F(MatlBuyPolicyTests, MixedActiveDormant) {
```

Comment:
Are these tests commented out on purpose?

Reply:
oops, no. at least, not anymore